### PR TITLE
[informatica-powercenter-soap] First integration of Powercenter SOAP API

### PR DESCRIPTION
**Pull request by ntorionbearstudio**

### CODE REVIEW

File: technologies/job/informatica-powercenter-soap/tasks/instanceActions.js [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      Timeout: DEFAULT_TIMEOUT,
+    });
+
+    const resBody = getResponseBodyFromSOAPRequest(res);
+
+    if (resBody && resBody.length > 0 && resBody[0]) {
+      const logs = resBody[0].Buffer;
+      const logsLines = logs.split('\n');
+
+      return Response.success(logsLines.map((logLine) => Log(logLine, Stream.STDOUT, null)));
```
Comment (yoannfleurydev) [inline technologies/job/informatica-powercenter-soap/tasks/instanceActions.js:128]:
```suggestion
      return Response.success(logsLines.map((logLine) => Log(logLine, Stream.STDOUT)));
```

No need to give `null`, this call contains a default value.

Comment (yoannfleurydev) [reply technologies/job/informatica-powercenter-soap/tasks/instanceActions.js:128]:
```suggestion
      return Response.success(logsLines.map((logLine) => Log(logLine)));
```

Same for the second parameter. If there is only STDOUT, no need to give it.

> https://github.com/saagie/sdk/blob/master/packages/sdk/src/sdk/Log.js#L10

File: technologies/job/informatica-powercenter-soap/workflows/instanceActions.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      Timeout: DEFAULT_TIMEOUT,
+    });
+
+    const resBody = getResponseBodyFromSOAPRequest(res);
+
+    if (resBody && resBody.length > 0 && resBody[0]) {
+      const logs = resBody[0].Buffer;
+      const logsLines = logs.split('\n');
+
+      return Response.success(logsLines.map((logLine) => Log(logLine, Stream.STDOUT, null)));
```
Comment (yoannfleurydev) [inline technologies/job/informatica-powercenter-soap/workflows/instanceActions.js:130]:
```suggestion
      return Response.success(logsLines.map((logLine) => Log(logLine)));
```

File: technologies/job/informatica-powercenter-soap/package.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  "private": true,
+  "scripts": {
+    "start": "saagie-sdk start",
+    "build": "saagie-sdk build",
+    "new:context": "saagie-sdk init"
+  },
+  "dependencies": {
+    "@saagie/sdk": "0.10.0",
+    "axios": "0.19.2",
+    "bluebird": "^3.5.5",
```
[the remark this answers is not in the document]
Comment (ntorionbearstudio) [reply technologies/job/informatica-powercenter-soap/package.json]: yes it was useless, I remove it

File: technologies/job/informatica-powercenter-soap/job-states.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const { JobStatus } = require('@saagie/sdk');
+
+export const JOB_STATES = {
+  ABORTED: JobStatus.KILED,
```
Comment (youenchene) [inline technologies/job/informatica-powercenter-soap/job-states.js:4]: Type on Job Status.KILED ==> JobStatus.KILLED

**Pull request merged**